### ISSUE TITLE
What is the point of the complex polar form?

### ISSUE THREAD

**Issue by Beakerboy** [markrogoyski/math-php#423]
I'm pretty sure for all Z, Z->polarForm() = Z. the polar form is just a way to use the abs and arg to decompose the complex number in a different way.

In your test you have
```
[5, 2, 5.3851648071 * COS(0.3805063771), 5.3851648071 *  SIN(0.3805063771)]
```

if you do the math, this is [5, 2, 5, 2]. This makes sense as an axiom test maybe, but I don't think anyone would ever actually call $z->polarForm(), since, in essence it just returns $Z with some extra floating point errors added in. I'm working on some tests for $z->exp() and noticed that function. It could make sense to have a factory method that creates a complex number from the polar form, maybe? `$Z = Complex::fromPolarForm($r, $θ);`

```php
public static function fromPolarForm($r, $θ): Complex
{
    return new Complex($r * \cos($θ), $r * \sin($θ));
}
```

**Comment by markrogoyski** [comment]
It was probably just low hanging fruit when I reviewed the Wikipedia entry on complex numbers. Looks like I added it in 2017. I don't remember =)

**Comment by Beakerboy** [comment]
I’d mark it deprecated for the next version since it seems useless.

**Comment by markrogoyski** [comment]
I think the current implementation might be wrong. I think it should be this:
```php
    public function polarForm(): Complex
    {
        $r = $this->abs();
        $θ = $this->arg();

        return new Complex($r, $θ);
    }
```

Whether or not it is useful or not, I cannot say, but for reference, Python offers this as part of the built-in standard library.
```python
In [134]: c = complex(5, 2)

In [135]: cmath.polar(c)
Out[135]: (5.385164807134504, 0.3805063771123649)
```

**Comment by Beakerboy** [comment]
I could be wrong, but my guess is it in not returning a complex number, but a list of the abs and arg.

**Comment by markrogoyski** [comment]
Yeah. It's a tuple, but yeah, an immutable list.
```python
In [136]: type(cmath.polar(c))
Out[136]: tuple
```

I'll change it to an array.

**Comment by markrogoyski** [comment]
I've changed this in the develop branch. Thanks for bringing this up.

**Comment by markrogoyski** [comment]
This will be fixed in the next release, v2.4.0. Thanks again.
